src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1472,9 +1472,15 @@ Connection connectInternal(Properties propsIn,
             sPropValue = activeConnectionProperties.getProperty(sPropKey);
             if (null != sPropValue) {
                 enclaveAttestationProtocol = sPropValue;
-                if (!AttestationProtocol.isValidAttestationProtocol(enclaveAttestationProtocol)) {
-                    throw new SQLServerException(SQLServerException.getErrString("R_enclaveInvalidAttestationProtocol"),
-                            null);
+                if (enclaveAttestationProtocol.equalsIgnoreCase("HGS")) {
+                    this.enclaveProvider = new SQLServerVSMEnclaveProvider();
+                } else if (enclaveAttestationProtocol.equalsIgnoreCase("AAS")) {
+                    this.enclaveProvider = new SQLServerAASEnclaveProvider();
+                } else {
+                    if (connectionlogger.isLoggable(Level.SEVERE)) {
+                        connectionlogger.severe(toString() + " "
+                                + SQLServerException.getErrString("R_enclaveInvalidAttestationProtocol"));
+                    }
```

Comment:
I think we should keep the enums in SQLServerDriver instead of hardcoding it here. It's similar to the other properties there like SSProtocol, ApplicationIntent, ColumnEncryption etc